### PR TITLE
Add webhook support (custom URL) as an option for notifications

### PR DESCRIPTION
**Pull request by ghost**
Fixes:
- Adds support for what is described in issue #2939 and feature [+475](http://feathub.com/SickRage/SickRage/+475), albeit using webhooks instead of a custom script placed on-disk

Proposed changes in this pull request:
- The webhook notifier sends a POST request to the configured URL using the JSON format in the following structure for snatched/downloaded episodes and subtitles:
```
{
    "title": "Notification title",
    "ep_name": "Episode name"
}
```
and the following format for update/login notifications:
```
{
    "title": "Notification title",
    "update_text": "SickRage updated to version X"
}
```
- The webhook notifier allows the user to configure the URL which is to receive the POST request
- The webhook notifier also allows the user to optionally configure a `token` value, which is sent as the value of the `X-SICKRAGE-AUTH` header during the request - this is for security on the request-receiving side

- [X] PR is based on the DEVELOP branch
- [X] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [X] Read [contribution guide](https://github.com/SickRage/SickRage/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Review by OmgImAlexis — CHANGES_REQUESTED** [review]
There should be no changes to the font files or the `vender.js` for this.

**Comment by ghost** [comment]
@OmgImAlexis Reverted other assets changes and only included `core.min.js`

**Review by OmgImAlexis — APPROVED** [review]
@sharkykh can you check over the Python changes.

File: gui/slick/views/config_notifications.mako [hunk 1]
```diff
@@ -2473,6 +2473,120 @@
...
                 </div>
             </div>
+
+            <div class="config-group-divider"></div>
+
+            <!-- /custom-notification component-group //-->
+            <div class="row">
+                <div class="col-lg-3 col-md-4 col-sm-4 col-xs-12">
+                    <div class="component-group-desc">
+                        <span class="icon-notifiers-webhook" title="${_('Webhook')}"></span>
```
Comment (sharkykh) [inline gui/slick/views/config_notifications.mako:2483]: We need to add an icon.
[This looks best](https://user-images.githubusercontent.com/10238474/27895421-5850981c-621b-11e7-84c0-5cd94f9fee3f.png). But the icons are [in a sprite file](https://github.com/SickRage/SickRage/blob/2f775438aad7325d37d393215c95dbe690b6b582/gui/slick/images/32px_sprite_colored_notifiers.png).
[The slack icon](https://github.com/SickRage/SickRage/blob/2f775438aad7325d37d393215c95dbe690b6b582/gui/slick/images/Slack_Mark_Web.png) also needs to be merged into that sprite. I'll see what I can do, later...

**Edit:** Done

File: sickbeard/notifiers/webhook.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            logger.log("Wrong data structure sent to webhook URL", logger.ERROR)
+            return False
+
+        return True
+
+    def notify_snatch(self, ep_name, title=notifyStrings[NOTIFY_SNATCH]):
+        if sickbeard.WEBHOOK_NOTIFY_ONSNATCH:
+            self._notifyWebhook({
+                "title": title,
+                "ep_name": ep_name
```
Comment (sharkykh) [inline sickbeard/notifiers/webhook.py:130]: IMO, all of the payloads should have the same keys - 
`title` is fine, just make the second one consistent, like `message`.
You may add `type`, if you want, and set it to the `NOTIFY_x` variables (they're integers)

File: sickbeard/notifiers/webhook.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with SickRage. If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function, unicode_literals
+
+import urlparse
```
Comment (sharkykh) [inline sickbeard/notifiers/webhook.py:22]: import that from `requests.compat` instead, 
for Python 2/3 compatibility (for the future)

**Comment by miigotu** [comment]
erm, we already have a webhook notifier, its just called something different and I forget what it is.....

**Comment by sharkykh** [comment]
@miigotu Slack?

**Comment by n0ncetonic** [comment]
Would love to have a generic webhook notification

**Comment by miigotu** [comment]
Both the discord and slack notifier can be used as a generic webhook ^
However, if the conflicts are taken care of I will review this and merge.

**Pull request closed without merging**